src/protobuf-net.Test/Issues/Issue742.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.IO;
+using Xunit;
+
+namespace ProtoBuf.Test.Issues
+{
+    public class Issue742
+    {
+        [Fact]
+        public void TestNullBasicTypeSerialization()
+        {
+            long? nullableLong = 11;
+            var serialized = Serialize<long?>(nullableLong);
```

Comment:
it would be good to check what the actual bytes are here; maybe an `out string hex` from `Serialize`, and `hex = BitConverter.ToString(stream.GetBuffer(), 0, (int)stream.Length);` - and especially if we're talking about a *change* in behaviour from 2.4.6, we need to check that the bytes are the **same** as they were in 2.4.6; they probably are, but we need to prove it

Reply:
Tried to implement your proposal